### PR TITLE
adjusts provisioning for remote execution ssh for non-root user

### PR DESCRIPTION
**Pull request by dgoetz**
I adjusted the provisioning for remote execution based on ssh for non-root user to get it running "out-of-the-box".
As I did not wanted to mess up other environments I added the possibility to create user if not existing by setting a global parameter and the creation of sudo files if effective user method is set to sudo.

Tested on Debian and SLES.
For RHEL I am not sure if requiretty is still default but disabling it for sure wont break anything.

### CODE REVIEW

**Comment by mmoll** [comment]
Please don't indent things inside ERB blocks, as the resulting, rendered shell output will have all the spaces in it. 👍 visually otherwise.

**Comment by dgoetz** [comment]
Removed the indention and force pushed the amended change.

File: snippets/remote_execution_ssh_keys.erb [hunk 1]
```diff
@@ -27,6 +32,12 @@
 user_exists=false
 getent passwd <%= ssh_user %> >/dev/null 2>&1 && user_exists=true
 
+<% if ssh_user != 'root' && @host.params['remote_execution_create_user'] -%>
```
Comment (mmoll) [inline snippets/remote_execution_ssh_keys.erb]: is remote_execution_create_user a boolean?

Comment (dgoetz) [reply snippets/remote_execution_ssh_keys.erb]: I used a global parameter set to true, so it is not actually a boolean but I used it as one. 
But I am open for other suggestions, if you prefer some other handling.

Comment (mmoll) [reply snippets/remote_execution_ssh_keys.erb]: I guess/hope `@host.param_true?('param_name')` also works for r_ex templates

**Comment by dgoetz** [comment]
Adjusted the boolean handling.

**Comment by mmoll** [comment]
@stbenjam could you have a look? `@host.param_true` can be used here as well, I hope...

**Comment by stbenjam** [comment]
Yea param_true should be available, this is just a normal provisioning template.

I didn't test it but looks good to me.

**Comment by mmoll** [comment]
allright, thanks @stbenjam!
merged, danke @dgoetz!

**Pull request merged**